Sources/Identity/CustomerInfo+ActiveDates.swift

```diff
@@ -48,7 +48,17 @@ extension CustomerInfo {
     }
 
     static func extractExpirationDates(_ subscriber: CustomerInfoResponse.Subscriber) -> [String: Date?] {
-        return subscriber.subscriptions.mapValues { $0.expiresDate }
+        var map = [String: Date?]()
+
+        for (productId, subscription) in subscriber.subscriptions {
+            if let productPlanIdentfier = subscription.productPlanIdentifier {
+                map["\(productId):\(productPlanIdentfier)"] = subscription.expiresDate
+            } else {
+                map[productId] = subscription.expiresDate
+            }
+        }
+
+        return map
```

Comment:
Nit, but you can make this functional too:
```suggestion
        return Dictionary(
            uniqueKeysWithValues: subscriber
                .subscriptions
                .lazy
                .map { productID, subscription in
                    let key: String
                    let value = subscription.expiresDate

                    if let productPlanIdentfier = subscription.productPlanIdentifier {
                        key = "\(productID):\(productPlanIdentfier)"
                    } else {
                        key = productID
                    }

                    return (key, value)
                }
        )
```

I think that's a bit more clear that every value is the same (`expiresDate`), and we just change the key pro subscriptions with a plan.